src/rust/graph-mutation/src/graph_mutation.rs

```diff
@@ -0,0 +1,551 @@
+        r_edge_name: EdgeName,
+    ) -> Result<(), GraphMutationManagerError> {
+        self.write_dropper
+            .check_edges(
+                tenant_id,
+                from_uid,
+                to_uid,
+                f_edge_name,
+                r_edge_name,
+                |f_edge_name, r_edge_name| async move {
+                    // todo: Batch statements are currently not supported by the Scylla rust client
+                    //       https://github.com/scylladb/scylla-rust-driver/issues/469
+                    let tenant_ks = tenant_keyspace_name(tenant_id);
+
+                    let f_statement = format!(
+                        r"
+                        INSERT INTO {tenant_ks}.edges (
+                            source_uid,
+                            destination_uid,
+                            f_edge_name,
+                            r_edge_name
+                        )
+                        VALUES (?, ?, ?, ?)
+                        ",
+                    );
+                    let r_statement = f_statement.clone();
+
+                    let mut batch: scylla::batch::Batch = Default::default();
+                    batch.statements.reserve(2);
+                    batch.append_statement(Query::from(f_statement));
+                    batch.append_statement(Query::from(r_statement));
+                    batch.set_is_idempotent(true);
+
+                    self.scylla_client
+                        .session
+                        .batch(
+                            &batch,
+                            (
+                                (
+                                    from_uid.as_i64(),
+                                    to_uid.as_i64(),
+                                    &f_edge_name.value,
+                                    &r_edge_name.value,
+                                ),
+                                (
+                                    to_uid.as_i64(),
+                                    from_uid.as_i64(),
+                                    &r_edge_name.value,
+                                    &f_edge_name.value,
+                                ),
+                            ),
+                        )
+                        .await?;
+                    Ok(())
+                },
+            )
+            .await
+    }
+}
+
+#[async_trait::async_trait]
+impl GraphMutationApi for GraphMutationManager {
+    type Error = GraphMutationManagerError;
+
+    /// Create Node allocates a new node in the graph, returning the uid of the new node.
+    #[tracing::instrument(skip(self), err)]
+    async fn create_node(
+        &self,
+        request: CreateNodeRequest,
+    ) -> Result<CreateNodeResponse, Self::Error> {
+        let uid = self
+            .uid_allocator_client
+            .allocate_id(request.tenant_id)
+            .await?;
+        let uid = Uid::from_u64(uid).ok_or_else(|| GraphMutationManagerError::ZeroUid)?;
```

Comment:
Seems like that `Uid` should store a `std::num::NonZeroU64` internally instead of a `u64`, but I get that's not for this PR.